imap_processing/codice/codice_l1a_lo_species.py

```diff
@@ -0,0 +1,286 @@
+                deltas,
+                dims=("epoch",),
+                attrs=cdf_attrs.get_variable_attributes(
+                    "epoch_delta_minus", check_schema=False
+                ),
+            ),
+        },
+        attrs=cdf_attrs.get_global_attributes(logical_source_id),
+    )
+    # Add first few unique variables
+    # TODO: add attrs in future PR
+    l1a_dataset["spin_period"] = xr.DataArray(
+        unpacked_dataset["spin_period"].values * constants.SPIN_PERIOD_CONVERSION,
+        dims=("epoch",),
+        attrs=cdf_attrs.get_variable_attributes("spin_period"),
+    )
+    l1a_dataset["k_factor"] = xr.DataArray(
+        np.array([constants.K_FACTOR]),
+        dims=("spin_sector",),
+        attrs=cdf_attrs.get_variable_attributes("k_factor"),
+    )
+    l1a_dataset["voltage_table"] = xr.DataArray(
+        np.array(voltage_data),
+        dims=("esa_step",),
+        attrs=cdf_attrs.get_variable_attributes("voltage_table", check_schema=False),
+    )
+    l1a_dataset["data_quality"] = xr.DataArray(
+        unpacked_dataset["suspect"].values,
+        dims=("epoch",),
+        attrs=cdf_attrs.get_variable_attributes("data_quality"),
+    )
+    l1a_dataset["acquisition_time_per_step"] = xr.DataArray(
+        calculate_acq_time_per_step(sci_lut_data["lo_stepping_tab"]),
+        dims=("esa_step",),
+        attrs=cdf_attrs.get_variable_attributes(
+            "acquisition_time_per_step", check_schema=False
+        ),
+    )
+
+    # Carry over these variables from unpacked data to l1a_dataset
+    l1a_carryover_vars = [
+        "sw_bias_gain_mode",
+        "st_bias_gain_mode",
+        "rgfo_half_spin",
+        "nso_half_spin",
+    ]
+    # Loop through them since we need to set their attrs too
+    for var in l1a_carryover_vars:
+        l1a_dataset[var] = xr.DataArray(
+            unpacked_dataset[var].values,
+            dims=("epoch",),
+            attrs=cdf_attrs.get_variable_attributes(var),
+        )
+
+    # Finally, add species data variables and their uncertainties
+    for idx, species in enumerate(species_names):
+        if view_tab_obj.apid == CODICEAPID.COD_LO_SW_SPECIES_COUNTS and species in [
+            "heplus",
+            "cnoplus",
+        ]:
+            species_attrs = cdf_attrs.get_variable_attributes("lo-pui-species-attrs")
+            unc_attrs = cdf_attrs.get_variable_attributes("lo-pui-species-unc-attrs")
+        else:
+            species_attrs = cdf_attrs.get_variable_attributes("lo-species-attrs")
+            unc_attrs = cdf_attrs.get_variable_attributes("lo-species-unc-attrs")
+
+        direction = (
+            "Sunward"
+            if view_tab_obj.apid == CODICEAPID.COD_LO_SW_SPECIES_COUNTS
+            else "Non-Sunward"
+        )
+        # Replace {species} and {direction} in attrs
+        species_attrs["CATDESC"] = (
+            species_attrs["CATDESC"]
+            .replace("{species}", species)
+            .replace("{direction}", direction)
```

Comment:
I think you can call the format function on these.
```suggestion
            species_attrs["CATDESC"]
            .format(species=species, direction=direction)
```

Reply:
so create a function in here and call it?

Reply:
No, it is already available, I think the comemnt I made should work.

Reply:
cool